src/components/DataDimension/DataDimension.js

```diff
@@ -1,21 +1,46 @@
 import { useConfig } from '@dhis2/app-runtime'
 import PropTypes from 'prop-types'
-import React from 'react'
+import React, {
+    createContext,
+    useContext,
+    useCallback,
+    useEffect,
+    useState,
+} from 'react'
+import {
+    dataTypeMap,
+    DIMENSION_TYPE_EXPRESSION_DIMENSION_ITEM,
+} from '../../modules/dataTypes.js'
 import { DIMENSION_ID_DATA } from '../../modules/predefinedDimensions.js'
 import ItemSelector from './ItemSelector.js'
 
+const DataDimensionCtx = createContext({})
+
 const DataDimension = ({
     onSelect,
     selectedDimensions,
     displayNameProp,
+    enabledDataTypes,
     infoBoxMessage,
     onCalculationSave,
+    visType,
 }) => {
     const { serverVersion } = useConfig()
-    const supportsEDI =
-        `${serverVersion.major}.${serverVersion.minor}.${
-            serverVersion.patch || 0
-        }` >= '2.40.0'
+
+    const filterDataTypesByVersion = useCallback(
+        (dataTypes) =>
+            dataTypes.filter(({ id }) =>
+                // Calculations only available from 2.40
+                id === DIMENSION_TYPE_EXPRESSION_DIMENSION_ITEM
```

Comment:
This is technically correct but personally I'd write this as `id !== DIMENSION_TYPE_EXPRESSION_DIMENSION_ITEM || serverVersion.minor >= 40` to avoid having to use `true`

Reply:
Refactored.